File: Adorable/app/src/main/java/merryweather/com/adorable/di/component/ActivityComponent.java
package merryweather.com.adorable.di.component;


import android.content.Context;

import dagger.Component;
import merryweather.com.adorable.MainActivity;
import merryweather.com.adorable.api.OpenWeatherMapService;
import merryweather.com.adorable.di.modules.ActivityModule;
import merryweather.com.adorable.di.scopes.PerActivity;

/**
 * Created by S on 17.05.2018.
 */

@PerActivity
@Component(dependencies = AppComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    Context context();

    OpenWeatherMapService openWeatherMapService();

    void inject(MainActivity activity);

}

File: Tech45/app/src/main/java/merryweather/com/tech45/di/component/ActivityComponent.kt
package merryweather.com.tech45.di.component


import android.content.Context

import dagger.BindsInstance
import dagger.Component
import merryweather.com.tech45.departments.DepartmentsActivity
import merryweather.com.tech45.di.factory.ViewModelFactory
import merryweather.com.tech45.di.modules.ViewModelModule
import merryweather.com.tech45.login.LoginActivity
import merryweather.com.tech45.api.TechService
import merryweather.com.tech45.di.modules.ActivityModule
import merryweather.com.tech45.di.scopes.PerActivity


/**
 * Created by S on 17.05.2018.
 */

@PerActivity
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(ActivityModule::class, ViewModelModule::class))
interface ActivityComponent {

    fun context(): Context

    fun techService(): TechService

    fun inject(activity: LoginActivity)

    fun inject(activity: DepartmentsActivity)
}
